Entities/Common/GUI/ActorHUDStartPos.as

```diff
@@ -14,12 +16,91 @@ f32 getHUDY()
 const f32 HUD_X = getHUDX();
 const f32 HUD_Y = getHUDY();
 
+bool hoverOnResupplyIcon(Vec2f icon_pos, Vec2f icon_size)
+{
+	Vec2f mouse_pos = getControls().getMouseScreenPos();
+
+	if (mouse_pos.x > icon_pos.x && mouse_pos.x < icon_pos.x + icon_size.x * 2
+		&& mouse_pos.y > icon_pos.y && mouse_pos.y < icon_pos.y + icon_size.y * 2)
+	{
+		return true;
+	}
+
+	return false;
+}
+
 Vec2f getActorHUDStartPosition(CBlob@ blob, const u8 bar_width_in_slots)
 {
 	f32 width = bar_width_in_slots * 40.0f;
 	return Vec2f(getHUDX() + 180 + 50 + 8 - width, getHUDY() - 40);
 }
 
+void DrawResupplyOnHUD(CBlob@ this, Vec2f tl)
+{
+	CPlayer@ p = this.getPlayer();
+	if (p is null) return;
+	
+	string name = this.getName();
+
+	GUI::SetFont("menu");
+
+	string propname = getCTFTimerPropertyName(p, "builder");
+
+	if (getRules().exists(propname)) 
```

Comment:
Early return avoids unnecessary indent: e.g. changing to `!if (getRules().exists(propname)) { return; }`

Reply:
fixed